### PR TITLE
[17.0][MIG] document_page_approval: Migration to 17.0

### PR DESCRIPTION
**Pull request by victoralmau**
Supersedes https://github.com/OCA/knowledge/pull/498

Migration to 17.0

Changes done:
- [x] Squash administrative commits
- [x] Purge translations
- [x] Remove `ir.model.access.csv` file https://github.com/OCA/knowledge/pull/498#discussion_r1832184917
- [x] Improve tests (change to setUpClass)
- [x] `document_page`: Replace `name_get()` to `_compute_display_name()` https://github.com/OCA/knowledge/pull/498#discussion_r1832179917
- [x] Remove `readonly=1` (redudant, it already is) defined in `approved_uid` and `approved_date` fields in views
- [x] Restore `readonly=1` to the state field
- [x] Remove duplicate _Back to draft_ button

Please @pilarvargas-tecnativa and @carlos-lopez-tecnativa can you review it?

@Tecnativa TT51529

### CODE REVIEW

File: document_page_approval/views/document_page_approval.xml [hunk 1]
```diff
@@ -174,26 +171,26 @@
...
                     string="Change Requests"
                     type="object"
-                    attrs="{'invisible':[('has_changes_pending_approval','=',False),('user_has_drafts','=',False)]}"
+                    invisible="not has_changes_pending_approval and not user_has_drafts"
                     icon="fa-edit"
                 />
             </xpath>
             <field name="content_uid" position="after">
-                <field name="approved_uid" />
+                <field name="approved_uid" readonly="1" />
```
Comment (pilarvargas-tecnativa) [inline document_page_approval/views/document_page_approval.xml]: Are all these new readonly needed?

**Review by pilarvargas-tecnativa — APPROVED** [review]
Code review :+1:

**Review by max3903 — APPROVED** [review]

File: document_page_approval/models/document_page.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        res = super()._create_history(vals)
+        res.action_to_approve()
+        return res
+
+    def action_changes_pending_approval(self):
+        self.ensure_one()
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "document_page_approval.action_change_requests"
+        )
+        context = literal_eval(action["context"])
```
Comment (carlos-lopez-tecnativa) [inline document_page_approval/models/document_page.py]:
```
  File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/mnt/data/odoo-addons-dir/document_page_approval/models/document_page.py", line 141, in action_changes_pending_approval
    context = literal_eval(action["context"])
  File "/opt/odoo/odoo/tools/_monkeypatches.py", line 78, in literal_eval
    return orig_literal_eval(expr)
  File "/usr/lib/python3.10/ast.py", line 64, in literal_eval
    node_or_string = parse(node_or_string.lstrip(" \t"), mode='eval')
  File "/usr/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 2
    
IndentationError: unexpected indent

```
image

**Review by carlos-lopez-tecnativa — CHANGES_REQUESTED** [review]

**Review by carlos-lopez-tecnativa — APPROVED** [review]

**Pull request merged**